depthai_ros_driver/CMakeLists.txt

```diff
@@ -65,15 +68,21 @@ include_directories (SYSTEM
     ${catkin_INCLUDE_DIRS}
 )
 
-# internal library, not to be exported
-# add_library (${PROJECT_NAME}_lib src/depthai_base.cpp)
-# target_link_libraries (${PROJECT_NAME}_lib
-#     ${catkin_LIBRARIES}
-#     ${OpenCV_LIBRARIES}
-#     ${Boost_LIBRARIES}
-#     depthai-core
-# )
-# add_dependencies (${PROJECT_NAME}_lib ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+add_library (${PROJECT_NAME}_pipeline
+    src/pipelines/preview_pipeline.cpp
+    src/pipelines/stereo_pipeline.cpp
+    src/pipelines/mobilenet_ssd_pipeline.cpp
+)
+target_link_libraries (${PROJECT_NAME}_pipeline
+    ${catkin_LIBRARIES}
+    ${OpenCV_LIBRARIES}
+    ${Boost_LIBRARIES}
+    depthai-core
```

Comment:
SYSTEM INTERFACE

Reply:
I thought SYSTEM can only be added in include_ and not link_?

Reply:
Oh, my mistake. Edited

Reply:
I removed unnecessary dependencies so that INTERFACE is not necessary.

Reply:
INTERFACE is a keyword for CMake to show what visibility the dependencies have. Without that, the behavior of CMake can change from version to version (due to deprecated features)

Reply:
`target_link_libraries (${PROJECT_NAME}_pipelines INTERFACE` causes include errors in `depthai-core` headers. 

`PRIVATE` instead of `INTERFACE`?
https://stackoverflow.com/a/64843062